### PR TITLE
Add /api/manifest.yml and accept API key on /api/index.html.

### PR DESCRIPTION
**Pull request by soulshake**
## Release Playbook
- [ ] Rebase against master
- [ ] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

**Comment by ddollar** [comment]
Would like to have manifest.yml pulled from convox/rack on Github if at all possible. Changes to that file are going to be part of PRs on that repo.

**Comment by soulshake** [comment]
@ddollar How would you feel about hosting manifest.yml on site temporarily? That file is going to need a lot of changes, and it'll be slower to update on the Rack repo, won't it? Do you have another idea?

**Comment by ddollar** [comment]
if it’s possible with cors or whatever i’d like to have the manifest in rack.. i think prs there are the right primitive

if it's too hard then putting it in site is ok for now

**Comment by soulshake** [comment]
I'm sure it should be simple: All we need is an `Origin` header when requesting the github url, e.g. 

`curl -i -H "Origin: http://convox.com"` + 
* `https://raw.githubusercontent.com/convox/rack/master/api/manifest.yml` or
* `-X GET https://api.github.com/repos/convox/rack/contents/api/manifest.yml` for json (though unauthenticated requests are limited to 60 per hour)

but I don't see how to do that in the `new SwaggerUi()` call.

TL;DR: I'd love if we could merge this as it is but only as a temporary measure, hopefully. Unless someone else can figure it out in the meantime.

**Comment by soulshake** [comment]
Follow-up to my own earlier questions, there's some good info [on this page](https://github.com/swagger-api/swagger-editor/blob/master/docs/cors.md).

For example:

`To run Chrome with HTTP access control disabled use --disable-web-security flag.`

To be honest, I don't think we would be risking much by setting `Access-Control-Allow-Origin` to `*`, at least until we have the mechanisms in place to allow users to create Convox API apps and provide a domain that they'll be making calls from. But doesn't hurt to be on the safe side for now.

**Comment by ddollar** [comment]
If we set `Access-Control-Allow-Origin` to `*` we open up a potential XSS vulnerability. If I can get you to load some arbitrary Javascript it can make calls to `https://console.convox.com` on your behalf, including deleting a Rack.

**Comment by soulshake** [comment]
I understand the reasoning, but at the same time, we want people to build things with our API. Is it common for most APIs to have such restrictions? Either way, the ones that do provide an interface for users to say what domain their app will be making calls from. By preventing other origins, we're making our API unusable in a browser context, aren't we?
